pkg/k8s/watchers/cilium_egress_gateway_policy.go

```diff
@@ -0,0 +1,130 @@
+	_, egpController := informer.NewInformer(
+		cache.NewListWatchFromClient(ciliumNPClient.CiliumV2alpha1().RESTClient(),
+			"ciliumegressnatpolicies", v1.NamespaceAll, fields.Everything()),
+		&cilium_v2alpha1.CiliumEgressNATPolicy{},
+		0,
+		cache.ResourceEventHandlerFuncs{
+			AddFunc: func(obj interface{}) {
+				var valid, equal bool
+				defer func() { k.K8sEventReceived(metricCENP, metricCreate, valid, equal) }()
+				if cenp := k8s.ObjToCENP(obj); cenp != nil {
+					valid = true
+					err := k.addCiliumEgressNATPolicy(cenp)
+					k.K8sEventProcessed(metricCENP, metricCreate, err == nil)
+				}
+			},
+			UpdateFunc: func(oldObj, newObj interface{}) {
+				var valid, equal bool
+				defer func() { k.K8sEventReceived(metricCENP, metricUpdate, valid, equal) }()
+
+				oldCenp := k8s.ObjToCENP(oldObj)
+				if oldCenp == nil {
+					return
+				}
+				deleteErr := k.deleteCiliumEgressNATPolicy(oldCenp)
+				newCenp := k8s.ObjToCENP(newObj)
+				if newCenp == nil {
+					return
+				}
+				valid = true
+				addErr := k.addCiliumEgressNATPolicy(newCenp)
+				k.K8sEventProcessed(metricCENP, metricUpdate, deleteErr == nil && addErr == nil)
+			},
+			DeleteFunc: func(obj interface{}) {
+				var valid, equal bool
+				defer func() { k.K8sEventReceived(metricCENP, metricDelete, valid, equal) }()
+				cenp := k8s.ObjToCENP(obj)
+				if cenp == nil {
+					return
+				}
+				valid = true
+				err := k.deleteCiliumEgressNATPolicy(cenp)
+				k.K8sEventProcessed(metricCENP, metricDelete, err == nil)
+			},
+		},
+		k8s.ConvertToCiliumEgressNATPolicy,
+	)
+
+	k.blockWaitGroupToSyncResources(
+		wait.NeverStop,
+		nil,
+		egpController.HasSynced,
+		k8sAPIGroupCiliumEgressNATPolicyV2,
+	)
+
+	go egpController.Run(wait.NeverStop)
+	k.k8sAPIGroups.AddAPI(k8sAPIGroupCiliumEgressNATPolicyV2)
+}
+
+func (k *K8sWatcher) addCiliumEgressNATPolicy(cenp *cilium_v2alpha1.CiliumEgressNATPolicy) error {
+	scopedLog := log.WithFields(logrus.Fields{
+		logfields.CiliumEgressNATPolicyName: cenp.ObjectMeta.Name,
+		logfields.K8sUID:                    cenp.ObjectMeta.UID,
+		logfields.K8sAPIVersion:             cenp.TypeMeta.APIVersion,
+	})
+
+	ep, err := egresspolicy.Parse(cenp)
+	if err != nil {
+		scopedLog.WithError(err).Warn("Failed to add CiliumEgressNATPolicy: malformed policy config.")
+		return err
+	}
+	if _, err := k.egressPolicyManager.AddEgressPolicy(*ep); err != nil {
+		scopedLog.WithError(err).Warn("Failed to add CiliumEgressNATPolicy.")
+		return err
+	}
+
+	scopedLog.Info("Added CiliumEgressNATPolicy")
+	return err
+}
+
+func (k *K8sWatcher) deleteCiliumEgressNATPolicy(cenp *cilium_v2alpha1.CiliumEgressNATPolicy) error {
+	scopedLog := log.WithFields(logrus.Fields{
+		logfields.CiliumEgressNATPolicyName: cenp.ObjectMeta.Name,
+		logfields.K8sUID:                    cenp.ObjectMeta.UID,
+		logfields.K8sAPIVersion:             cenp.TypeMeta.APIVersion,
+	})
+
+	ep, err := egresspolicy.Parse(cenp)
```

Comment:
there's no need to parse a `cenp` on deletes because we only need to get its ID. I would suggest to create a function to generate the ID from a `cenp` and use that ID to pass down to the `DeleteEgressPolicy`

Reply:
Fixed. thx.